src/charm.py

```diff
@@ -74,13 +64,12 @@ def _pwgen(length=None):
             # A random length is ok to use a weak PRNG
             length = random.choice(range(35, 45))
         alphanumeric_chars = [
-            letter for letter in (string.ascii_letters + string.digits)
-            if letter not in 'l0QD1vAEIOUaeiou']
+            letter for letter in (string.ascii_letters + string.digits) if letter not in 'l0QD1vAEIOUaeiou'
```

Comment:
I'm curious. What's the point of excluding those characters?

Reply:
I have no idea, I ported this https://github.com/juju/charm-helpers/blob/master/charmhelpers/core/host.py#L949 in, as I had dependency issues with packing the charm with the refactoring I did. I wondered as well, but didn't explore further.